### PR TITLE
WPEFramework -> Thunder

### PR DESCRIPTION
**Pull request by wouterlucas**
- Rename recipes to Thunder / Thunder Nano Services respectively
- Adjust RDEPS/DEPS and recipe-core images/package groups
- Upstreamed wpebackend-rdk backend selector from Morty
- Bumps to latest Webkit & Thunder

**DO NOT MERGE** has issues with `wpe-westeros-image` build, broken virtual input implementation. `wpe-eglfs-image` is tested and validated. Raising for review/awareness and will update once westeros image is resolved/tested again.

### CODE REVIEW

File: recipes-wpe/thunder/include/thunder.inc [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # WPE Framework common
 HOMEPAGE = "https://github.com/WebPlatformForEmbedded"
-SECTION = "wpe"
+SECTION = "thunder"
```
Comment (kraj) [inline recipes-wpe/thunder/include/thunder.inc]: this in not known subsystem maybe call it plugins or some such

File: recipes-wpe/thunder/include/compositor.inc [hunk 2]
```diff
@@ -8,22 +8,28 @@ WPE_COMPOSITOR_HW_READY 	?= "0"
...
+WPE_COMPOSITOR_IRMODE		?= "16"
+WPE_COMPOSITOR_MEMORY_GFX	?= ""
 WPE_COMPOSITOR_OUTOFPROCESS ?= "true"
 WPE_COMPOSITOR_RESOLUTION	?= "1080p"
-WPE_COMPOSITOR_MEMORY_GFX	?= ""
-WPE_COMPOSITOR_IRMODE		?= "16"
 WPE_COMPOSITOR_SVP			?= "None"
 
+# change to Weston or Westeros, will be included in case of Wayland
+WPE_COMPOSITOR_SUB_IMPLEMENTATION = "Westeros"
```
Comment (kraj) [inline recipes-wpe/thunder/include/compositor.inc]: making it case insensitive would be better

File: recipes-wpe/thunder/include/webkitbrowser.inc [hunk 3]
```diff
@@ -12,6 +12,10 @@ WEBKITBROWSER_DISKCACHE ?= "0"
 WEBKITBROWSER_XHRCACHE ?= "false"
 WEBKITBROWSER_TRANSPARENT ?= "false"
 WEBKITBROWSER_THREADEDPAINTING ?= "2"
+WEBKITBROWSER_RESOLUTION = "1080p"
+
+# Raspberry PI 3 has 4 cores, set to 4 for RPIs
+WEBKITBROWSER_THREADEDPAINTING_rpi ?= "4"
```
Comment (kraj) [inline recipes-wpe/thunder/include/webkitbrowser.inc]: this means older pi will suffer please document pi3 being minimum needed

File: recipes-wpe/thunder/thunder_git.bb [hunk 4]
```diff
@@ -1,41 +1,44 @@
-SUMMARY = "Web Platform for Embedded Framework"
+SUMMARY = "Thunder is a tiny embedded services layer"
 HOMEPAGE = "https://github.com/WebPlatformForEmbedded"
-SECTION = "wpe"
+SECTION = "thunder"
 LICENSE = "Apache-2.0"
 LIC_FILES_CHKSUM = "file://LICENSE;md5=1fe8768cbb5fd322f7d50656133549de"
 PR = "r0"
```
Comment (kraj) [inline recipes-wpe/thunder/thunder_git.bb]: remove PR while here

File: recipes-wpe/thunder/thunder_git.bb [hunk 5]
```diff
@@ -1,41 +1,44 @@
...
            "
-SRCREV = "e770ecf4ba1f7e9293c5c0eac7fe4b660025c389"
+SRCREV = "7e1dcb188c74cb4c24881bba23e6cdbe91daaaf1"
 
 inherit cmake pkgconfig systemd update-rc.d
 
 # Yocto root is under /home/root
-WPEFRAMEWORK_PERSISTENT_PATH = "/home/root"
-WPEFRAMEWORK_SYSTEM_PREFIX = "OE"
+THUNDER_PERSISTENT_PATH = "/home/root"
```
Comment (kraj) [inline recipes-wpe/thunder/thunder_git.bb]: these settings should be overridable

File: recipes-wpe/thunder/include/thunder.inc (hunk 1 above)
Comment (wouterlucas) [reply recipes-wpe/thunder/include/thunder.inc]: Okay, plugins sounds great. I dont know what else to fill in here.

File: recipes-wpe/thunder/include/compositor.inc (hunk 2 above)
Comment (wouterlucas) [reply recipes-wpe/thunder/include/compositor.inc]: Yeah this is the way the CMAKE is setup at the other end. We can ofc send a pull request to the main thunder repo to make this lowercase.

File: recipes-wpe/thunder/include/webkitbrowser.inc (hunk 3 above)
Comment (wouterlucas) [reply recipes-wpe/thunder/include/webkitbrowser.inc]: I guess only RPI0, as RPI1 won't work anymore with ARMv6 to begin with. But sure will add a comment. 

Alternatively is there a way to determine this from the machine features/config?

File: recipes-wpe/thunder/thunder_git.bb (hunk 5 above)
Comment (wouterlucas) [reply recipes-wpe/thunder/thunder_git.bb]: Good one, thanks. Will update.

**Comment by kraj** [comment]
@wouterlucas will you post a v2 ?

**Comment by wouterlucas** [comment]
I hope so! It's still on my list, just swamped at the moment.

**Pull request closed without merging**